static/js/pm_list.js

```diff
@@ -76,35 +133,143 @@ export function update_dom_with_unread_counts(counts) {
     set_count(counts.private_message_count);
 }
 
-function should_expand_pm_list(filter) {
-    const op_is = filter.operands("is");
+export function update_collapsed_private_messages_section() {
+    // when we need to update the collapsed private messages section, we first check
+    // for any active conversation thread which is supposed to be present outside to it.
+    const conversations = pm_list_data.get_conversations();
+    const active_conversation = conversations.find((conversation) => conversation.is_active);
+
+    if (active_conversation) {
+        // If we find an active conversation thread satisfying the requirement, then we update our
+        // DOM with it's node.
+        const node = [pm_list_dom.keyed_pm_li(active_conversation)];
+        const new_dom = pm_list_dom.pm_ul(node);
+        const $container = ui.get_content_element($("#sidebar-private-messages-container"));
+        function replace_content(html) {
+            $container.html(html);
+        }
 
-    if (op_is.length >= 1 && op_is.includes("private")) {
-        return true;
+        function find() {
+            return $container.find("ul");
+        }
+        vdom.update(replace_content, find, new_dom, prior_dom);
+        prior_dom = new_dom;
+    } else {
+        // else we make the list empty and mark the section as collapsed.
+        $(".expanded_private_messages").empty();
```

Comment:
you don't need to call `$(".expanded_private_messages").empty();`, that happens in `close()`

Reply:
Ah yes!
Thanks for this, updated the PR with it.